### PR TITLE
Adds process for zero touch registration and functionality to update processes and configs 

### PR DESCRIPTION
**Pull request by Mehul-Kumar-27**
## Release Notes

- **New Features**
	- Introduced a new Docker Compose configuration for PostgreSQL and pgAdmin services.
	- Added a command-line interface for managing container runtime configurations, including `containerd` and `CRI-O`.
	- Implemented a robust event broker system for managing process events.
	- Added a comprehensive configuration management system for jobs and state processes.
	- Enhanced the configuration management for satellite systems with new constants and structured data handling.
	- Introduced a new function for creating runtime directories and improved file handling utilities.

- **Bug Fixes**
	- Enhanced error handling and logging across various components, including configuration fetching and event publishing.

- **Refactor**
	- Restructured configuration management to improve clarity and maintainability.
	- Streamlined the main application flow by simplifying the execution logic.
	- Refactored the `Satellite` struct and related methods for improved configuration handling and process scheduling.
	- Removed obsolete methods and types to simplify the codebase.

- **Chores**
	- Updated dependencies and versions in the project configuration.

### CODE REVIEW

File: cmd/container_runtime/read_config.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+)
+
+func NewReadConfigCommand(runtime string) *cobra.Command {
+	readContainerdConfig := &cobra.Command{
+		Use:   "read",
+		Short: fmt.Sprintf("Reads the config file for the %s runtime", runtime),
+		PersistentPreRun: func(cmd *cobra.Command, args []string) {
+			utils.SetupContextForCommand(cmd)
+			config.InitConfig()
+		},
```
[the remark this answers is not in the document]
Comment (Mehul-Kumar-27) [reply cmd/container_runtime/read_config.go:19]: resolved in container pr

File: value/io.containerd.metadata.v1.bolt/meta.db
[the remark this answers is not in the document]
Comment (bupd) [reply value/io.containerd.metadata.v1.bolt/meta.db:1]: cc // @Vad1mo

[the remark this answers is not in the document]
Comment (Mehul-Kumar-27) [reply value/io.containerd.metadata.v1.bolt/meta.db:1]: I don't think there is any need for any binary file in this PR. Will check how this binary was generated and will add this is gitignore.

File: internal/state/artifact.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+	Repository string   `json:"repository,omitempty"`
+	Tags       []string `json:"tag,omitempty"`
+	Labels     []string `json:"labels"`
+	Type       string   `json:"type,omitempty"`
+	Digest     string   `json:"digest,omitempty"`
+	Deleted    bool     `json:"deleted"`
+	Name       string   `json:"name,omitempty"`
+}
+
+// NewArtifact creates a new Artifact object
```
Comment (bupd) [inline internal/state/artifact.go:31]: Remove ai gen comments.

**Review by bupd — CHANGES_REQUESTED** [review]
Great work overall! However, there are some opportunities for improvement.

**Always Explain Why, Not What** in comments.

The code can be much more concise by Removing unnecessary comments 

I see a lot of comments which don't make sense. Please remove them.

Maybe try making the code concise by not implementing methods to get the values of variables. Which already can be accessed directly.


> Good code explains HOW, Comment explains WHY.

I may be hard on this. but 100+ lines for just a simple GET request from Ground control feels too much.

Overall The code feels more verbose. Which can be fixed by following GO's best practices. 

Thanks

File: internal/state/artifact.go (hunk 2 above)
Comment (Mehul-Kumar-27) [reply internal/state/artifact.go]: Thanks for review @bupd , have made some changes based on your review.

**Pull request merged**